demo-apps/chaos-monkey-demo-app-unleash-toggles/src/main/java/com/example/chaos/monkey/chaosdemo/ChaosDemoApplication.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.chaos.monkey.chaosdemo;
+
+import de.codecentric.spring.boot.chaos.monkey.configuration.ChaosMonkeyProperties;
+import de.codecentric.spring.boot.chaos.monkey.configuration.toggles.ChaosToggleNameMapper;
+import javax.servlet.http.HttpServletRequest;
+import no.finn.unleash.Unleash;
+import no.finn.unleash.UnleashContext;
+import no.finn.unleash.UnleashContextProvider;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@SpringBootApplication
+public class ChaosDemoApplication {
+
+  public static void main(String[] args) {
+    SpringApplication.run(ChaosDemoApplication.class, args);
+  }
+
+  @Bean
+  public UnleashContextProvider unleashContextProvider() {
+    return () -> {
+      UnleashContext.Builder context = new UnleashContext.Builder();
+
+      try {
+        // This isn't a true production approach. Spring Security would typically be used instead.
+        // This is merely a basic 'fake user' for demo/testing
+        HttpServletRequest curRequest =
```

Comment:
Wouldn't it make sense to just use Spring Security in that demo app? I prefer real world use cases (as long as it doesnt add too much of complexity so that the purpose of the demo is lost)

I think the demo app need some polish (the casting isnt that nice, and the swallow of the exception etc..) and I think there too much duplicated tests (which most likely dont get any value) in that demo project.

Reply:
I've pulled in Spring Security and updated the examples.  I removed some of the tests. Several of those tests, like the ControllerTests aren't exercising any ChaosMonkey behavior.   Should I remove those? 

I'm tempted to update them to trigger chaos behaviors, so they can serve as an example to others, but if that isn't likely to be useful in actual use cases, I'm happy to delete those tests so they don't distract from the others.